app/main.py

```diff
@@ -1,13 +1,31 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
     pass
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
     pass
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [
+        {
+            "errors": [
+                format_linter_error(error) for error in errors
+            ],
+            "path": file_path,
+            "status": "failed" if errors else "passed"
```

Comment:
this code duplicates the function format_single_linter_file, it can be replaced with a function, passing arguments to it, and thereby reducing the function code format_linter_report